### ISSUE TITLE
Lecture 3 & Lecture 4: Issue 정리

### ISSUE THREAD

**Issue by InhyeokYoo** [InhyeokYoo/CS224N#2]
3주차:
- [x] window padding 
- [ ] label 값 byte 인코딩
- [ ] loss 계산
- [ ] likelihood, MLE, CE, logit, softmax
- [ ] logit, sigmoid, softmax의 관계: https://opentutorials.org/module/3653/22995

4주차:
- [ ] ReLU vs. Leaky ReLU
- [ ] annealing vs. decay
  - [ ] annealing: factor x

**Comment by abooundev** [comment]
* NER
- window padding 
- label 값 byte 인코딩
- loss 계산

**Comment by InhyeokYoo** [comment]
- logit, sigmoid, softmax의 관계: https://opentutorials.org/module/3653/22995

**Comment by abooundev** [comment]
> * NER
>   
>   * window padding:해결
>   * label 값 byte 인코딩: 찾는 중
>   * loss 계산
> *   Reshaping. Takes in a (B, L) LongTensor -> Outputs a (B, L~, S) LongTensor ~의미? 

1.  window padding
- 문장 앞뒤로 padding을 window_size 만큼 넣어준 이유는, 가장 앞, 뒤 단어도 중심 단어가 될 수 있도록 넣어준 것입니다.
- 실제로 input은 중심단어 + 문맥단어 = 5로 맞춰서 미니배치로 입력됩니다.

2. loss 계산
- 각 단어의 레이블 값에 바이트 인코딩 한 레이블값 포함  
1) raw label: tensor([0, 0, 0, 0, 1])
2) 배치 사이즈 x label 길이의 새로운 tensor 정의
tensor([[0., 0., 0., 0., 0.],
          [0., 0., 0., 0., 0.]])
3) 새로운 tensor에 원래 label값을 1행에 삽입
tensor([[0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 1.]])
4) 원래 label 값에 ~연산 및 바이트 인코딩 적용: ~l.byte()
l: tensor([0, 0, 0, 0, 1]
~연산 적용: tensor([-1, -1, -1, -1, -2])
바이트 인코딩: tensor([255, 255, 255, 255, 254], dtype=torch.uint8)
5) 새로운 tensor에  ~연산 적용한 원래 label값을 0행에 삽입
 tensor([[255., 255., 255., 255., 254.],
           [  0.,   0.,   0.,   0.,   1.]]))

- output layer(2개 노드)의 출력값: softmax 적용후 + negative log 적용
- 출력값과 변환된 레이블 행렬 간의 행렬 요소 곱
